src/participant.rs

```diff
@@ -300,7 +300,7 @@ pub(crate) trait InnerProtocolParticipant: ProtocolParticipant {
                 "Received a Ready message from {}, but Ready should only be sent to yourself!",
                 message.from()
             );
-            return Err(InternalError::ProtocolError);
+            return Err(InternalError::ProtocolError(None));
```

Comment:
If someone else sends us a `Ready` message, they're the ones breaking the protocol. This should be `message.from`.